lib/manageiq/api/common/user.rb

```diff
@@ -0,0 +1,63 @@
+module ManageIQ
+  module API
+    module Common
+      class User
+        IDENTITY_KEY = 'x-rh-identity'.freeze
+        VALID_USER_HEADER_KEYS = %w(account_number user username email is_org_admin)
+
+        def self.with_logged_in_user(user = nil)
+          yield Thread.current[:attr_current_user] = new(user)
```

Comment:
Not sure what the point of the thread current is, if you're not actually creating a direct accessor for it.

Is with_logged_in_user the only was to use this?  Can you give an example how this would work in a controller, because I can't envision how it will work

Reply:
@syncrou @Fryguy 

We are trying to model this after the User.with_user in ManageIQ so that setting of the Thread Local Storage (TLS) and unsetting of the TLS happens in a single function and a block gets passed in.

https://github.com/ManageIQ/manageiq/blob/0648257adeab79e217a4d7783b961108f113e797/app/models/user.rb#L263

So in the controllers for every action I am thinking we would have

def some_action
   User.with_logged_in_user do
      call_service_object
      render :json ...
  end
end